pkg/northbound/gnmi/set.go

```diff
@@ -132,15 +141,22 @@ func (s *Server) Set(ctx context.Context, req *gnmi.SetRequest) (*gnmi.SetRespon
 	return setResponse, nil
 }
 
-func (s *Server) doUpdateOrReplace(u *gnmi.Update, targetUpdates mapTargetUpdates) map[string]string {
+func (s *Server) doUpdateOrReplace(u *gnmi.Update, targetUpdates mapTargetUpdates) (map[string]*change.TypedValue, error) {
 	target := u.Path.GetTarget()
 	updates, ok := targetUpdates[target]
 	if !ok {
-		updates = make(map[string]string)
+		updates = make(map[string]*change.TypedValue)
 	}
 	path := utils.StrPath(u.Path)
-	updates[path] = utils.StrVal(u.Val)
-	return updates
+
+	var err error
```

Comment:
I do not believe you need a var here, just declare it inline

Reply:
Done